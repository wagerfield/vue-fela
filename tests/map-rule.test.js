import { createRenderer } from 'fela'
import { mapRule } from '../index'
import {
  testSnapshot,
  installPlugin,
  wrapComponent
} from './test-helpers'
import props from './component-props'
import rules from './component-rules'

const localVue = installPlugin()
const fela = createRenderer()
const rule = rules.border

const props1 = { outerRadius: 0, innerRadius: 0, active: false }
const props2 = { outerRadius: 4, innerRadius: 2, active: true }

const createComponent = (optProps) => ({
  props,
  render(createElement) {
    return createElement('div', {
      class: this.className
    })
  },
  computed: {
    className: mapRule(rule, optProps)
  }
})

describe('mapRule(rule, optProps)', () => {

  beforeEach(fela.clear)

  it('returns a function', () => {
    expect(mapRule(rule)).toEqual(expect.any(Function))
  })

  it('renders expected snapshots from propsData', () => {
    const component = createComponent()
    const wrapper = wrapComponent(component, localVue, fela)

    testSnapshot(wrapper)

    wrapper.setProps({ active: true })
    testSnapshot(wrapper)

    wrapper.setProps({ outerRadius: 4 })
    testSnapshot(wrapper)

    wrapper.setProps({ innerRadius: 2 })
    testSnapshot(wrapper)
  })

  it('renders expected snapshots from optProps', () => {
    testSnapshot(wrapComponent(createComponent(props1), localVue, fela))
    testSnapshot(wrapComponent(createComponent(props2), localVue, fela))
  })

  it('optProps takes precedence over propsData', () => {
    const component = createComponent(props1)
    const wrapper1 = wrapComponent(component, localVue, fela)
    const wrapper2 = wrapComponent(component, localVue, fela, props2)
    expect(wrapper1.vm.className).toBe(wrapper2.vm.className)
  })
})
