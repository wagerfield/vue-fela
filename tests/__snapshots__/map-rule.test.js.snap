// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapRule(rule, optProps) renders expected snapshots from optProps 1`] = `<div class="a b c"></div>`;

exports[`mapRule(rule, optProps) renders expected snapshots from optProps 2`] = `<div class="d e c"></div>`;

exports[`mapRule(rule, optProps) renders expected snapshots from propsData 1`] = `<div class="a b c"></div>`;

exports[`mapRule(rule, optProps) renders expected snapshots from propsData 2`] = `<div class="a d c"></div>`;

exports[`mapRule(rule, optProps) renders expected snapshots from propsData 3`] = `<div class="e d c"></div>`;

exports[`mapRule(rule, optProps) renders expected snapshots from propsData 4`] = `<div class="f d c"></div>`;
